workers/ohsome_quality_analyst/indicators/mapping_saturation/models.py

```diff
@@ -0,0 +1,428 @@
+from ohsome_quality_analyst.indicators.mapping_saturation import metrics
+
+
+class BaseStatModel(ABC):
+    """Base Statistical Model"""
+
+    def __init__(self, xdata: np.ndarray, ydata: np.ndarray):
+        assert np.shape(xdata) == np.shape(ydata)
+        self.xdata = xdata
+        self.ydata = ydata
+        self.coefficients = {}
+        self.fitted_values = None
+        self.asym_conf_int = None
+
+    @property
+    @abstractmethod
+    def name(self) -> str:
+        """Name of the statistical model"""
+        pass
+
+    @property
+    @abstractmethod
+    def function_formula(self) -> str:
+        """Function formula of the statistical model"""
+        pass
+
+    @property
+    @abstractmethod
+    def asymptote(self) -> str:
+        pass
+
+    @property
+    def mae(self) -> str:
+        """Mean absolute error"""
+        return metrics.mae(self.ydata, self.fitted_values)
+
+    def as_dict(self):
+        return {
+            key: getattr(self, key)
+            for key in (
+                "name",
+                "function_formula",
+                "asymptote",
+                "mae",
+                "xdata",
+                "ydata",
+                "coefficients",
+                "fitted_values",
+            )
+        }
+
+
+class Sigmoid(BaseStatModel):
+    """Sigmoid model.
+
+    Function formula:
+
+        f(x) = L / (1 + e^(-k * (x - x_0)))
+
+    Function parameters:
+
+        L, the curve's maximum value/asymptotic (the plateaus);
+        k, the Logistic growth rate or steepness of the curve;
+        x_0, the x value of the sigmoid's midpoint (inflection point);
+
+    Function formula is taken from the Wikipedia article "Logistic function":
+    https://en.wikipedia.org/wiki/Logistic_function
+    """
+
+    name = "Sigmoid model"
+    function_formula = "f(x) = L / (1 + e^(-k * (x - x_0)))"
+
+    def __init__(self, xdata, ydata):
+        super().__init__(xdata, ydata)
+        # curve_fit: Use non-linear least squares to fit a function, f, to data.
+        # popt: Optimal values for the parameters as array
+        # pcov: The estimated covariance of popt
+        popt, pcov = curve_fit(
+            self.function,
+            xdata=xdata,
+            ydata=ydata,
+            p0=self.initial_guess(),
+            bounds=self.bounds(),
+        )
+
+        self.coefficients = {
+            "x_0": popt[0],
+            "k": popt[1],
+            "L": popt[2],
+        }
+        self.asym_conf_int = self.confint(popt, pcov)
+        self.fitted_values = self.function(xdata, **self.coefficients)
+
+    def function(
+        self, x: np.float64, x_0: np.float64, k: np.float64, L: np.float64
+    ) -> np.float64:
+        """Sigmoid function."""
+        return L / (1 + np.exp(-k * (x - x_0)))
+
+    def initial_guess(self) -> tuple:
+        """Get an initial guess on parameters for single Sigmoid function."""
+        x_0 = self.xdata.size / 2
+        k = 0
+        L = self.ydata.max()
```

Comment:
- [ ] For `ndarray.max()`: set initial value
- [ ] capital `L`

Reply:
I would like to use variable names as they are used in the function definition. In this case capital `L`.
This is also setup to be ignored setup.py for flake8 and sonar conf for sonarcloud.
https://github.com/GIScience/ohsome-quality-analyst/blob/a8dd694b4909e89fdf3e101f2458aa341be92eaf/sonar-project.properties#L21

Reply:
I set the initial values of all the min/max functions to zero.